Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -222,9 +251,29 @@ protected override void ProcessRecord()
 
             // Content type to use for the new object.
             string objContentType = null;
-
             Stream contentStream = null;
-            if (!string.IsNullOrEmpty(File))
+
+            if (ParameterSetName == ParameterSetNames.ContentsFromFolder)
+            {
+                // User gives us the path to a folder, we will resolve the path and upload the contents of that folder.
+                // Have to take care of / and \ in the end of the directory path because Path.GetFileName will return
+                // empty string if that is not trimmed off.
+                string resolvedFolderPath = GetFullPath(Folder).TrimEnd("/\\".ToCharArray());
+                if (string.IsNullOrWhiteSpace(resolvedFolderPath) || !Directory.Exists(resolvedFolderPath))
+                {
+                    throw new DirectoryNotFoundException($"Directory '{resolvedFolderPath}' cannot be found.");
+                }
+
+                string gcsObjectNamePrefix = Path.GetFileName(resolvedFolderPath);
+                if (!string.IsNullOrWhiteSpace(ObjectNamePrefix))
+                {
+                    gcsObjectNamePrefix = Path.Combine(ObjectNamePrefix, gcsObjectNamePrefix);
+                }
+                UploadDirectory(resolvedFolderPath, metadataDict, ConvertLocalToGcsFolderPath(gcsObjectNamePrefix));
```

Comment:
Seems like we might want to add a progress indicator, but that can wait for a future PR. A TODO or NOTE comment seems appropriate.

````
// NOTE: There man be many (many) files being uploaded. Consider doing a scan ahead of time
// and putting up a progress indicator as the objects are created.
````

or 

````
// TODO(quoct): Add a progress indicator if there are many files.
````